src/Storefront/Page/Robots/Parser/RobotsDirectiveParser.php

```diff
@@ -0,0 +1,169 @@
+                    $issue = $unknownDirectiveEvent->issue;
+                    if ($issue === null) {
+                        $issue = new ParseIssue(
+                            $lineNumber,
+                            $line,
+                            'Unknown directive type: \'' . $directiveType . '\'',
+                            ParseIssueSeverity::WARNING
+                        );
+                    }
+                    $issues[] = $issue;
+                }
+
+                continue;
+            }
+
+            // Handle User-agent directive
+            if ($directiveTypeEnum === RobotsDirectiveType::USER_AGENT) {
+                // If we have a current block with directives, save it
+                if (\count($currentUserAgents) > 0 && \count($currentDirectives) > 0) {
+                    foreach ($currentUserAgents as $userAgent) {
+                        $userAgentBlocks[] = new RobotsUserAgentBlock($userAgent, $currentDirectives);
+                    }
+                    // Reset for new block
+                    $currentUserAgents = [];
+                    $currentDirectives = [];
+                }
+
+                // Add user agent to current block
+                $currentUserAgents[] = $directiveValue;
+
+                continue;
+            }
+
+            // Handle other directives
+            $directive = new RobotsDirective($directiveTypeEnum, $directiveValue);
+
+            if (\count($currentUserAgents) > 0) {
+                // We're in a user-agent block
+                $currentDirectives[] = $directive;
+            } else {
+                // Orphaned directive (backward compatibility)
+                if ($directiveTypeEnum->isPathBased()) {
+                    $orphanedPathDirectives[] = $directive;
+                } else {
+                    $issues[] = new ParseIssue(
+                        $lineNumber,
+                        $line,
+                        'Directive \'' . $directiveTypeEnum->value . '\' found outside user-agent block and will be ignored',
+                        ParseIssueSeverity::WARNING
+                    );
+                }
+            }
+        }
+
+        // Save last block if any
+        if (\count($currentUserAgents) > 0 && \count($currentDirectives) > 0) {
+            foreach ($currentUserAgents as $userAgent) {
+                $userAgentBlocks[] = new RobotsUserAgentBlock($userAgent, $currentDirectives);
+            }
+        }
+
+        $parsedResult = new ParsedRobots($userAgentBlocks, $orphanedPathDirectives, $issues);
+
+        // Dispatch parsing event to allow modifications
+        $parsingEvent = new RobotsDirectiveParsingEvent($text, $parsedResult, $context, $salesChannelId);
+        $this->eventDispatcher->dispatch($parsingEvent);
+
+        return $parsingEvent->parsedResult;
+    }
+
+    /**
+     * @deprecated tag:v6.8.0 - reason:becomes-obsolete - will be removed, as string parsing is deprecated
```

Comment:
`reason:becomes-obsolete` does not exists as exception for the deprecation PHPStan rule. 

the `reason:*` stuff is only used for methods, where we cannot add a call of `Feature::triggerDeprecationOrThrow(`, but which we have here. so this can simply be omitted. Teach that your AI please :wink: 

```suggestion
     * @deprecated tag:v6.8.0 - Will be removed with next major, as string parsing is deprecated
```

Reply:
Done. Also, the Rules have been updated.